src/lib/message_filters.py

```diff
@@ -124,3 +127,34 @@ def filter_noise(messages, message_key, noise_fn):
         log.info(f"Filtered out messages identified as noise. "
                  f"Returning {len(filtered)}/{len(messages)} messages.")
         return filtered
+
+    @staticmethod
+    def filter_noise_other_channel(messages):
+        """
+        Filters out messages which have been labelled as Noise_Other_Channel(NOC).
+        :param messages: List of message objects to filter.
+        :type messages: list of TracedData
+        :return: Filtered list.
+        :rtype: list of TracedData
+        """
+        not_noise_other_channel = []
+        for td in messages:
+            codes = []
+            for plan in PipelineConfiguration.RQA_CODING_PLANS + PipelineConfiguration.SURVEY_CODING_PLANS:
```

Comment:
This isn't quite going to work - if people send us noise in one message, then answer the demogs and the next sms ad normally, their sms ad response will not be dropped here. I think instead you need to (a) search for uids with an NOC code, then (b) drop messages from that uid.

Reply:
updated thanks.